index.js

```diff
@@ -1,9 +1,23 @@
 const contacts = require('bindings')('contacts.node')
 
-function getContactsByName(name) {
-  if (typeof name !== 'string') throw new TypeError('name must be a string')
+function getAllContacts(extraProperties = []) {
+  if (extraProperties.constructor.name !== 'Array') {
```

Comment:
Is `Array.isArray(extraProperties)` a better check for this?

Reply:
The latter is perhaps used more, but they're both correct. I'm ambivalent though 🤷‍♀️